### PR TITLE
Added missing server commands and updated compiling everything

### PR DESCRIPTION
**Pull request by Fisico**
- Added missing server commands.
- Splitted compiling everything documentation in separate files.
- updated compiling everything to correspond to Teeworlds 0.7.1 and bam 0.5.1.

Someone with a mac should test, if my documentation is correct.

should fix: #16, #14, #13 and #12

### CODE REVIEW

File: compiling_everything_linux.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    - `$ ./make_unix.sh`
+    - `$ cd ..`
+
+
+#### Compiling
+2. `cd /teeworldsSource`
+    - Changes to the teeworlds source directory
+3. `./bam/bam config`
+    - Runs bam configuration
+4. `./bam/bam conf=release -f`
```
Comment (oy) [inline compiling_everything_linux.md]: -f isn't necessary. should be mentioned as optional

File: compiling_everything_linux.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    - Compiles teeworlds (Client and Server)
+    - Flag `-f` will force a recompile
+    - Available targets:
```
Comment (oy) [inline compiling_everything_linux.md]: targets are: game, client, server, tools, versionserver, masterserver, content

debug and release have to be specified by "conf=xxx"

File: compiling_everything_linux.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        - `release` (for all in release mode)
+        - `debug` (for all in debug mode)
+        - `server_release`
+        - `server_debug`
+        - `client_release`
+        - `client_debug`
+    - To build the tools and master server in release mode use the following arguments:
+        - `conf=release tools masterserver`
+    - You may specify the architecture using `arch`, e.g. `arch=X86` or `arch=x64`.
```
Comment (oy) [inline compiling_everything_linux.md]: 'x86' and 'x86_64'

**Comment by Fisico** [comment]
@oy ok now?

**Comment by Dune-jr** [comment]
Thanks, that's some good work.

This got rid of the instructions to compile 0.6. I don't think that is a good idea to remove them this early - the page used to keep instructions on how to compile any previous version, afaik.

The paths are a bit messed up, e.g.: 
```5. The compiled game is located at `/teeworldsSource/build/..```

that's absolute paths. Probably better to just keep the zip nomenclature too: `./teeworlds-0.7.1` instead of `teeworldsSource`

**Comment by Fisico** [comment]
Thank you for your feedback.
Some 0.6 and previous documentation is already there, but I will add a bit more.

I don't think it is a good idea to name it ./teeworlds-0.7.1 because I also linked the GitHub repo and this version is not teeworlds-0.7.1 and will never be named like that.

Everyone that needs to compile teeworlds should know that /teeworldsSource/ is the Directory where the source is located. If the user does not know that, he is probably not familiar with cmd, shell, VS etc... And he will also not be able to hack the source. He will probably download the executable and starts playing.

**Comment by Dune-jr** [comment]
> I don't think it is a good idea to name it ./teeworlds-0.7.1 because I also linked the GitHub repo and this version is not teeworlds-0.7.1 and will never be named like that.

Then, to keep it generic, could you use `teeworlds-version`? I think the old doc did exactly that. Sorry, it's just nitpicking :)

File: compiling_everything_linux.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        - `server`
+        - `client`
+        - `content`
+        - `masterserver`
+        - `tools`
+    - To build the tools and master server in release mode use the following arguments:
+        - `conf=release tools masterserver`
+    - Flag `-f` will force a recompile
+    - You may specify the architecture using `arch`, e.g. `arch=X86` or `arch=x86_64`.
+5. The compiled game is located at `/teeworldsSource/build/..`
```
Comment (Dune-jr) [inline compiling_everything_linux.md]: Avoid the use of `..`, that goes to the parent folder. Use `/teeworldsSource/build/<arch>` or just `/teeworldsSource/build/`

File: compiling_everything_linux.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    - g++
+    - python
+    - alsa (asound)
+    - gl
+    - glu
+    - x11
+    - libsdl
+    - freetype
+2. Download and unzip [Teeworlds-source](https://github.com/teeworlds/teeworlds/releases) or [Teeworlds-latest-source](https://github.com/teeworlds/teeworlds)
+3. Download and unzip [bam](https://github.com/matricks/bam/archive/v0.5.1.tar.gz) to `\teeworldsSource\bam`
```
Comment (Dune-jr) [inline compiling_everything_linux.md]: The explanations for the 0.6 and earlier versions put `bam` next to `teeworlds` (probably saner). Better keep it uniform. Commands lower use `../bam/`

File: compiling_everything_linux.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    - gcc
+    - g++
+    - python
+    - alsa (asound)
+    - gl
+    - glu
+    - x11
+    - libsdl
+    - freetype
+2. Download and unzip [Teeworlds-source](https://github.com/teeworlds/teeworlds/releases) or [Teeworlds-latest-source](https://github.com/teeworlds/teeworlds)
```
Comment (Dune-jr) [inline compiling_everything_linux.md]: I would say either use English (Teeworlds source, Teeworlds latest source) or keep the nomenclature from the downloads (teeworlds-version, etc).

File: compiling_everything_linux.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+3. Download and unzip [bam](https://github.com/matricks/bam/archive/v0.5.1.tar.gz) to `\teeworldsSource\bam`
+4. Compiling bam  
+    - `$ cd bam`
+    - `$ ./make_unix.sh`
+    - `$ cd ..`
+
+
+#### Compiling
+2. `cd /teeworldsSource`
```
Comment (Dune-jr) [inline compiling_everything_linux.md]: This won't work, `/` makes you go to the root, remove it

File: compiling_everything_linux.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
++ `server_release`
++ `server_debug`
++ `client_release`
++ `client_debug`
+
+E.g. to build server debug use the following arguments:
+
+`../bam/bam server_debug`
+
+**Also:** Teeworlds 0.5.2 and earlier requires python 2.x to compile. Python 3.x will not work. Python 3.x support is introduced with Teeworlds 0.6.0.
```
Comment (Dune-jr) [inline compiling_everything_linux.md]: "Also" made sense under "Note". Better replace the first Also with "Note" here

File: compiling_everything_linux.md [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+#### Setup
+1. Use your package manager (apt-get, emerge or whatever is used on your distribution) to install the following (you will need the header files):
+    - gcc
+    - g++
+    - python
+    - alsa (asound)
+    - gl
+    - glu
+    - x11
+    - libsdl
```
Comment (Dune-jr) [inline compiling_everything_linux.md]: libsdl2 just for more emphasis on that

File: compiling_everything_linux.md [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+#### Compiling
+2. `cd /teeworldsSource`
+    - Changes to the teeworlds source directory
+3. `./bam/bam config`
+    - Runs bam configuration
+4. `./bam/bam conf=release`
+    - Compiles teeworlds (Client and Server)
+    - `conf=debug` will build the debug version
+    - Available targets for release and debug:
+        - `game`
```
Comment (Dune-jr) [inline compiling_everything_linux.md]: > - `game` (default)

Or use another way to indicate that game is built if no target is given

File: compiling_everything_mac.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+**Q: What is BAM?**
+
+Bam is the [build system made by matricks](http://matricks.github.io/bam/) used in Teeworlds.
+
+#### Setup
+1. Install Xcode from the Appstore or install Xcode Command Line Tools `xcode-select --install`. 
+3. Install libsdl using [brew](https://brew.sh/) `brew install sdl2`
+4. Install Freetype using [brew](https://brew.sh/) `brew install freetype`
+5. Download and unzip [Teeworlds-source](https://github.com/teeworlds/teeworlds/releases) or [Teeworlds-latest-source](https://github.com/teeworlds/teeworlds)
+6. Download and unzip [bam](https://github.com/matricks/bam/archive/v0.5.1.zip) to `\teeworldsSource\bam`
```
Comment (Dune-jr) [inline compiling_everything_mac.md]: I think Mac uses `/`

File: compiling_everything_mac.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+7. Compiling bam  
+    - `$ cd bam`
+    - `$ ./make_unix.sh`
+    - `$ cd ..`
+
+
+#### Compiling
+2. `cd /teeworldsSource`
```
Comment (Dune-jr) [inline compiling_everything_mac.md]: Again, that / is erroneous

File: compiling_everything_windows.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+### Setup (Using Microsoft Tools)
+1. Download and install [Visual Studio 2017 Community](https://visualstudio.microsoft.com/de/downloads/).
+
+
+# Windows x64 
+#### Setup
+1. Download and unzip [Teeworlds-source](https://github.com/teeworlds/teeworlds/releases) or [Teeworlds-latest-source](https://github.com/teeworlds/teeworlds)
+2. Download and install [Python 3.x](https://www.python.org/download/)
+3. Download and unzip [bam](https://github.com/matricks/bam/archive/v0.5.1.zip) to `\teeworldsSource\bam`
+    - Run `make_win64_msvc.bat` to compile BAM
```
Comment (Dune-jr) [inline compiling_everything_windows.md]: No caps for bam, there and below

File: compiling_everything_windows.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+
+# Windows x86 
+#### Setup
+1. Download and unzip [Teeworlds-source](https://github.com/teeworlds/teeworlds/releases) or [Teeworlds-latest-source](https://github.com/teeworlds/teeworlds)
+2. Download and install [Python 3.x](https://www.python.org/download/)
+3. Download and unzip [bam](https://github.com/matricks/bam/archive/v0.5.1.zip) to `\teeworldsSource\bam`
+    - Run `make_win32_msvc.bat` to compile BAM
+4. Download and unzip [SDL 2.0.8](https://www.libsdl.org/release/SDL2-devel-2.0.8-VC.zip)
+    - Copy `include` and `lib` to \teeworldsSource\other\sdl
+    - Copy `SDL2.lib` and `SDL2.main.lib` from `\teeworldsSource\other\sdl\lib\x86` in the root directory`\teeworldsSource`
```
Comment (Dune-jr) [inline compiling_everything_windows.md]: It's SDL2main.lib (replace elsewhere too)

**Comment by Dune-jr** [comment]
> Some of the comments apply to many places, like the / and \ issue, parse and replace.

The paths are still broken everywhere, @Fisico 

If you place / at the beginning of a path, it means the root.

**Pull request merged**